t/t2807-dump-cmd.t

```diff
@@ -0,0 +1,172 @@
+#!/bin/sh
+
+test_description='Test flux dump/restore'
+
+. $(dirname $0)/sharness.sh
+
+test_under_flux 1 minimal
+
+QUERYCMD="flux python ${FLUX_SOURCE_DIR}/t/scripts/sqlite-query.py"
+
+countblobs() {
+	$QUERYCMD -t 100 $(flux getattr content.backing-path) \
+		"select count(*) from objects;" | sed -e 's/.*= //'
+}
+
+test_expect_success 'flux-dump with no args prints Usage message' '
+	test_must_fail flux dump 2>dump-noargs.out &&
+	grep "Usage" dump-noargs.out
+'
+test_expect_success 'flux-restore with no args prints Usage message' '
+	test_must_fail flux restore 2>restore-noargs.out &&
+	grep "Usage" restore-noargs.out
+'
+test_expect_success 'flux-dump with no backing store fails' '
+	test_must_fail flux dump --checkpoint foo.tar 2>nostore.err &&
+	grep "No service matching kvs-checkpoint.get" nostore.err
+'
+test_expect_success 'flux-dump with bad archive file fails' '
+	test_must_fail flux dump /badfile.tar 2>badfile.err &&
+	grep badfile.tar badfile.err
+'
+test_expect_success 'load content-sqlite' '
+	flux module load content-sqlite
+'
+test_expect_success 'flux-dump --checkpoint with missing checkpoint fails' '
+	test_must_fail flux dump --checkpoint foo.tar 2>nocheck.err &&
+	grep "error fetching checkpoint" nocheck.err
+'
+
+test_expect_success 'load kvs and create some kvs content' '
+	printf "%-.*d" 100 0 >x.val &&
+	flux module load kvs &&
+	flux kvs put --no-merge a.b.c=testkey &&
+	flux kvs link linkedthing y &&
+	flux kvs put --no-merge x=$(cat x.val) &&
+	flux kvs link --target-namespace=smurf otherthing z
+'
+test_expect_success 'unload kvs' '
+	flux module remove kvs
+'
+test_expect_success 'dump default=kvs-primary checkpoint' '
+	flux dump --checkpoint foo.tar
+'
+test_expect_success 'repeat dump with -q' '
+	flux dump -q --checkpoint foo.tar
+'
+test_expect_success 'repeat dump with -v' '
+	flux dump -v --checkpoint foo.tar
+'
+test_expect_success 'repeat dump with --no-cache' '
+	flux dump --no-cache --checkpoint foo.tar
+'
+test_expect_success 'unload content-sqlite' '
+	flux content flush &&
+	flux content dropcache &&
+	flux module remove content-sqlite
+'
+# Small values are contained in dirent rather than separate objects, so expect:
+#  rootdir-v0 + (rootdir-v1 + a + b + x) + rootdir-v2 + rootdir-v3 + rootdir-v4
```

Comment:
it took me a while to grok this comment and get to how you go to 8.  Perhaps more clear (I think the +x above is a cut & paste error?)

```
rootdir-v0 (initial rootdir) + (rootdir-v1 (add a) + dir a + dir b) + (rootdir-v2 (add y)) + (rootdir-v3 (add x) + valref x) + (rootdir-v4 (add z)) = 8 blobrefs
```

Reply:
sorry, those comments weren't updated when the script changed!  I'll update and expand the comment so the casual reader won't be totally mystified!